From 2495c53afd1ba5b33605fd88ba0eef2aeb711bf0 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Fri, 13 May 2016 22:58:50 +0200
Subject: [PATCH 35/78] ar71xx: use correct PLL configuration register bitmask
 for QCA956x SoC.

Incorrect value causes clock inaccuracy as huge as 1/60.

Signed-off-by: Dmitry Ivanov <dima@ubnt.com>
Signed-off-by: Felix Fietkau <nbd@openwrt.org>

Backport of OpenWrt r47363
---
 .../patches-3.18/735-MIPS-ath79-add-support-for-QCA956x-SoC.patch     | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/target/linux/ar71xx/patches-3.18/735-MIPS-ath79-add-support-for-QCA956x-SoC.patch b/target/linux/ar71xx/patches-3.18/735-MIPS-ath79-add-support-for-QCA956x-SoC.patch
index b5c88e9..d3a14b2 100644
--- a/target/linux/ar71xx/patches-3.18/735-MIPS-ath79-add-support-for-QCA956x-SoC.patch
+++ b/target/linux/ar71xx/patches-3.18/735-MIPS-ath79-add-support-for-QCA956x-SoC.patch
@@ -529,7 +529,7 @@
 +#define QCA956X_PLL_CPU_CONFIG1_NFRAC_L_SHIFT		0
 +#define QCA956X_PLL_CPU_CONFIG1_NFRAC_L_MASK		0x1f
 +#define QCA956X_PLL_CPU_CONFIG1_NFRAC_H_SHIFT		5
-+#define QCA956X_PLL_CPU_CONFIG1_NFRAC_H_MASK		0x3fff
++#define QCA956X_PLL_CPU_CONFIG1_NFRAC_H_MASK		0x1fff
 +#define QCA956X_PLL_CPU_CONFIG1_NINT_SHIFT		18
 +#define QCA956X_PLL_CPU_CONFIG1_NINT_MASK		0x1ff
 +
@@ -541,7 +541,7 @@
 +#define QCA956X_PLL_DDR_CONFIG1_NFRAC_L_SHIFT		0
 +#define QCA956X_PLL_DDR_CONFIG1_NFRAC_L_MASK		0x1f
 +#define QCA956X_PLL_DDR_CONFIG1_NFRAC_H_SHIFT		5
-+#define QCA956X_PLL_DDR_CONFIG1_NFRAC_H_MASK		0x3fff
++#define QCA956X_PLL_DDR_CONFIG1_NFRAC_H_MASK		0x1fff
 +#define QCA956X_PLL_DDR_CONFIG1_NINT_SHIFT		18
 +#define QCA956X_PLL_DDR_CONFIG1_NINT_MASK		0x1ff
 +
-- 
2.10.1

